### PR TITLE
fix: [CDS-50192]: Fixed issue of Wizard form getting submitted even if validate is failing

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

This issues was happening due to type="submit" for the submit button is WizardFooter as we have also added onClick handler for this button and validating the pipeline related fields inside the onClick callback. So even if the validation was failing form was getting submit as formik was submitting the form due to type="submit".

Now added `event.preventDefault()` for submit button callback function and form submission has happening only due the button callback function.

#### Screenshots
Before:

https://user-images.githubusercontent.com/110591046/211649490-6b9ef0ab-3667-4d56-a036-7be295925460.mov

After:

https://user-images.githubusercontent.com/110591046/211649512-c12f524f-cfc6-4c4d-91b0-464cd5dcb011.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cd

**Comment by pankajsingh-harness** [comment]
retrigger coverage

File: src/modules/10-common/components/Wizard/WizardFooter.tsx [hunk 1]
```diff
@@ -111,7 +111,6 @@ export const WizardFooter = ({
           text={submitLabel || getString('submit')}
           variation={ButtonVariation.PRIMARY}
           rightIcon="chevron-right"
-          type="submit"
```
Comment (udayvunnam) [inline src/modules/10-common/components/Wizard/WizardFooter.tsx:114]: Can we keep the `type="submit"` and have the submit handler on the form instead? Mainly due to a11y that an enter doesn't submit form

Comment (pankajsingh-harness) [reply src/modules/10-common/components/Wizard/WizardFooter.tsx:114]: Added `type="submit"` back and added `event.preventDefault()` on `onClick` handler.

**Comment by pankajsingh-harness** [comment]
retrigger jest

**Review by vikrant-harness — APPROVED** [review]

**Review by udayvunnam — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**